File: app/src/main/java/com/example/knight/doublecheck/app/DoubleCheckTool.java
package com.example.knight.doublecheck.app;

import android.util.Log;

public class DoubleCheckTool {

    public static long CLICK_DURING_MILLIS = 300L * 1000000;
    private static final String TAG = DoubleCheckTool.class.getSimpleName();

    private static long lastClickTime = 0;

    public static boolean isClickable() {
        Log.i("liyachao","isClickable");
        long now = System.nanoTime();
        long last = lastClickTime;
        if (last == 0 || (now - last) > CLICK_DURING_MILLIS) {
            Log.i("liyachao","true: "+ ((now - last)));
            lastClickTime = System.nanoTime();

            return true;
        } else {
            Log.i("liyachao","false");
            return false;
        }
    }

}

File: knight-doubleCheck-library/src/main/kotlin/com/knight/doublecheck/library/DoubleCheckTool.kt
package com.knight.doublecheck.library

object DoubleCheckTool {
    private const val CLICK_DURING_MILLIS = 500L * 1000000L
    private var lastClickTime = 0L

    @JvmStatic
    fun isClickable(): Boolean {
        val now = System.nanoTime()
        val last = lastClickTime
        if (last == 0L || ((now - last) > CLICK_DURING_MILLIS)) {
            lastClickTime = System.nanoTime()

            return true
        } else {
            return false
        }
    }
}